internal/cache/util/util.go

```diff
@@ -180,3 +183,97 @@ func TruncateAndRemoveFile(filePath string) error {
 	}
 	return nil
 }
+
+// GetMemoryAlignedBuffer creates a buffer([]byte) of size bufferSize aligned to
+// memory address in multiple of alignSize.
+func GetMemoryAlignedBuffer(bufferSize uint64, alignSize uint64) (buffer []byte, err error) {
+	if bufferSize == 0 {
+		return make([]byte, 0), nil
+	}
+	if alignSize == 0 {
+		return make([]byte, bufferSize), nil
+	}
+
+	// Create and align buffer
+	createAndAlignBuffer := func() {
+		buffer = make([]byte, bufferSize+alignSize)
+		l := uint64(uintptr(unsafe.Pointer(&buffer[0])) % uintptr(alignSize))
+		skipOffset := alignSize - l
+		buffer = buffer[skipOffset : skipOffset+bufferSize]
+
+		// Check if buffer is aligned or not
+		l = uint64(uintptr(unsafe.Pointer(&buffer[0])) % uintptr(alignSize))
+		if l != 0 {
+			buffer = nil
+			err = fmt.Errorf("failed to align buffer")
+		}
+	}
+
+	// Though we haven't seen any error while aligning buffer but still it is safer
+	// to attempt few times in case alignment fails.
+	for try := 0; try < 3; try++ {
+		createAndAlignBuffer()
```

Comment:
make this function return buffer and err. Its confusing you are calling a nested function which is setting your return params